pkg/erofs/erofs_unsafe.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2023 The gVisor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package erofs
+
+import (
+	"unsafe"
+
+	"gvisor.dev/gvisor/pkg/errors/linuxerr"
+)
+
+func (i *Image) pointerAt(off uint64) unsafe.Pointer {
+	return unsafe.Pointer(uintptr(unsafe.Pointer(&i.bytes[0])) + uintptr(off))
+}
+
+type scalar interface {
+	uint8 | uint16 | uint32 | uint64
+}
```

Comment:
You can just use `Unsigned` constraint? https://pkg.go.dev/golang.org/x/exp/constraints#Unsigned

Reply:
Sure, will use it.